language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -347,20 +347,24 @@ genDefDataType curPkgId conName mod tpls def =
                      ["daml.STATIC_IMPLEMENTS_SERIALIZABLE_CHECK<" <> conName <> ">(" <> conName <> ")"]
             in ((typeDesc, serDesc), Set.unions $ map (Set.setOf typeModuleRef . snd) bs)
         DataEnum enumCons ->
-          let
-            typeDesc =
-                [ "export enum " <> conName <> " {"] ++
-                [ "  " <> cons <> " = " <> "\'" <> cons <> "\'" <> ","
-                | VariantConName cons <- enumCons] ++
-                [ "}"
-                , "daml.STATIC_IMPLEMENTS_SERIALIZABLE_CHECK<" <> conName <> ">(" <> conName <> ")"
-                ]
-            serDesc =
-                ["() => jtv.oneOf<" <> conName <> ">" <> "("] ++
-                ["  jtv.constant(" <> conName <> "." <> cons <> ")," | VariantConName cons <- enumCons] ++
-                [");"]
+          let cs = map unVariantConName enumCons
+              typeDesc =
+                let td = ["  " <> "'" <> cons <> "'" | cons <- cs]
+                    td' = reverse td
+                in [""] <> reverse (head td' : map (<> " | ") (tail td'))
```

Comment:
```suggestion
                "" : ["  | '" <> cons <> "'" | cons <- cs]
```
TypeScript seems to be perfectly happy to put a `|` at the beginning as well and that’s even used in the docs for unions and it makes your code much more readable.

Reply:
Very well spotted. Thanks.